### PR TITLE
docs: Enable dark mode and follow user preference

### PR DESCRIPTION
**Pull request by mortenn**
Not just an aestethic change, some people with certain vision problems are helped by dark mode.

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -203,4 +203,18 @@ theme:
...
+        icon: material/brightness-7
+        name: Switch to dark mode
+
+    # Palette toggle for dark mode
+    - media: "(prefers-color-scheme: dark)"
+      scheme: slate
+      primary: teal
+      toggle:
+        icon: material/brightness-4
+        name: Switch to light mode
```
Comment (morey-tech) [inline mkdocs.yml:220]: Fix spacing and remove comments to [pass the codegen action](https://github.com/argoproj/argo-cd/actions/runs/4274222709/jobs/7440682373).
```suggestion
  - media: "(prefers-color-scheme: light)"
    scheme: default
    primary: teal
    toggle:
      icon: material/brightness-7
      name: Switch to dark mode
  - media: "(prefers-color-scheme: dark)"
    scheme: slate
    primary: teal
    toggle:
      icon: material/brightness-4
      name: Switch to light mode
```

**Review by morey-tech — CHANGES_REQUESTED** [review]
Thanks for this! Minor change to pass the PR checks

File: mkdocs.yml [hunk 2]
```diff
@@ -203,4 +203,15 @@ theme:
...
     primary: teal
+    toggle:
+      icon: material/brightness-7
+      name: Switch to dark mode
+  - media: "(prefers-color-scheme: dark)"
+    scheme: slate
+    primary: teal
+    toggle:
+      icon: material/brightness-4
+      name: Switch to light mode
```
Comment (crenshaw-dev) [inline mkdocs.yml:217]:
```suggestion
  - media: "(prefers-color-scheme: light)"
    primary: teal
    scheme: default
    toggle:
      icon: material/brightness-7
      name: Switch to dark mode
  - media: "(prefers-color-scheme: dark)"
    primary: teal
    scheme: slate
    toggle:
      icon: material/brightness-4
      name: Switch to light mode
```

**Review by crenshaw-dev — APPROVED** [review]

File: mkdocs.yml [hunk 3]
```diff
@@ -203,4 +203,15 @@ theme:
...
+    scheme: default
+    toggle:
+      icon: material/brightness-7
+      name: Switch to dark mode
+  - media: "(prefers-color-scheme: dark)"
+    primary: teal
+    scheme: slate
+    toggle:
+      icon: material/brightness-4
+      name: Switch to light mode
```
Comment (crenshaw-dev) [inline mkdocs.yml:217]:
```suggestion
  - media: '(prefers-color-scheme: light)'
    primary: teal
    scheme: default
    toggle:
      icon: material/brightness-7
      name: Switch to dark mode
  - media: '(prefers-color-scheme: dark)'
    primary: teal
    scheme: slate
    toggle:
      icon: material/brightness-4
      name: Switch to light mode
```

**Comment by 34fathombelow** [comment]
@mortenn Thank you so much for this contribution.  It makes a night and day difference 😜

**Pull request merged**